assgnment2/funcions.dart

```diff
@@ -0,0 +1,183 @@
+
+import 'dart:convert';
+import 'util.dart';
+import 'user.dart';         /// two classes approach (util input, util validation)
+import 'dart:io';
+bool validateUserChoice(int userChoice){
+
+ if( userChoice <1 || userChoice > 5){
+  print('please enter an integer ranging 1-5');
+  return false;
+ }
+ return true;
+
+}
+ int takeUserChoice() {
+
+ int validChoice = Util.validateInteger();// kya ye valid
+
+ while(validateUserChoice(validChoice) == false){
+   validChoice = Util.validateInteger();
+ }
+ return validChoice;
+
+}
+
+void printOptions() {
+
+  print('Enter the as per options below');
+  print('1:For Adding user details');
+  print('2:Displayng the existing user detaiils');
+  print('3:Deleting the user details');
+  print('4:Saving the user details');
+  print('5:Exit');
+  
+}
+int validRollNumber(){
+  
+  int valid= Util.validateInteger();
+
+  while(User.regieteredRollNumber.contains(valid)){
+    print('please enter a unique roll number');
+    valid = Util.validateInteger();
+  }
+   return valid; 
+}
+
+
+ List<String> readCources(){
+ Set<String> courcesOffered = {"A" , "B" ,"C" , "D","E" , "F"};  
+ Set<String> currentCources = {};
+
+ print('Cources offered are A,B,C,D,E,F');
+
+
+ while(currentCources.length < 4){
+  print("please  select atleat 4 cources out of 6");
+  String cource = stdin.readLineSync()!;
+                                                                  
+  if(courcesOffered.contains(cource)){
+
+  if( currentCources.contains(cource)){
+    print('Cource already exist enter another on from offeredCources');
+  }
+  else
+  currentCources.add(cource);
+ }
+ else{
+  print('entered cource is not being offered');
+ }
+
+}
+print('choosen cources are');
+List<String> courceChosen=[];
+
+for(var curr in currentCources){
+  print(curr);
+  courceChosen.add(curr);
+}
+
+
+return courceChosen;
+}
+
+User readDeatais(){
+  String userName ='';
+  int userRollNumber= 0;
+  int userAge = 0;
+  String userAddress='';
+
+  print('enter the name of user');
+  userName = stdin.readLineSync()!;
+
+  print('enter the Roll number of user');
+
+  userRollNumber= validRollNumber();
+  
+
+  print('enter the address of user');
+  userAddress = stdin.readLineSync()!;
+
+  print('enter the Age of user');
+  userAge = Util.validateInteger();
+
+  List<String> userCources = readCources();
+  
+  return User(userName: userName, userAge: userAge, userAddress: userAddress, userRollNumber: userRollNumber, userCources: userCources);
+}
+
+void displayUser(){
```

Comment:
There should be 2 options in the programs - One to  display all users (just name and roll number) then another one to get all the details of a particular student.